Belonit's "animations for ares additions" corrections

CCCP84 opened the pull request:
I removed the duplications and brought the numbering to a single standard for YR and RA2. However, it seems to me that most of the remaining lines also do not make sense:

I checked theese and they do not provide any visible animations in the game:

1200=D
1202=gtpowexp
1203=tstlexp
1208=BBBLELRG
1211=xxxx
1212=TWLT024

And judging by the names, some look like kind of nonsense ("D" "xxxx")
DBRIS8LGDBRS10LG i deleted because it's obvious just a forgotten comma.

I have a suspicion that "tstlexp"(used in blow of Soviet power plant) is actually "KTSTLEXP", but the forgotten letter "K" 
It produces an explosion with electric lighting. Now it used for Kirov Elite bomb-animation. I checked it at the power plant, and it looks quite natural. Among the explosions, there's a flash of blue light, like an electrical fault. I think that's what they meant. It's probably just a typo.

In any case, it seems to me that the above lines are simply meaningless.

TWLT024 is found only in this place. It is not found anywhere else in any files and it was added only for RA2

I'll repeat once again that I left these lines and only deleted the duplication.

CCCP84 committed:
Belonit's "animations for ares additions" corrections

CCCP84 committed:
Update rulesmd.ini

Belonit commented:
> and brought the numbering to a single standard for YR and RA2.

Are you sure this is safe? Especially for campaign missions?

CCCP84 commented:
> Are you sure this is safe? Especially for campaign missions?

In my experience, the numbering sequence has no meaning at all, only the indexes after the "=" matter.

Belonit commented:
> DBRIS8LGDBRS10LG i deleted because it's obvious just a forgotten comma.

Yes, that's a missing comma, but it wasn't me, it was WESTWOOD, I added these animations to the list because the vanilla game adds them automatically, but ares disables automatic adding
<img width="816" height="500" alt="image" src="https://github.com/user-attachments/assets/658fc700-61a0-4d2d-9994-28db4f7dce78" />
<img width="834" height="339" alt="image" src="https://github.com/user-attachments/assets/e3ec075d-4dc8-4f3f-b90a-180951f735ee" />

CCCP84 commented:
CRNUKEWH needs to be fixed too

Belonit commented:
> In my experience, the numbering sequence has no meaning at all, only the indexes after the "=" matter.

Read this -https://modenc.renegadeprojects.com/Animations
I also know that animations in map triggers are set by internal index, not by name.

CnCRAZER commented:
> > In my experience, the numbering sequence has no meaning at all, only the indexes after the "=" matter.
> 
> Read this -https://modenc.renegadeprojects.com/Animations I also know that animations in map triggers are set by internal indices, not by name.

This is true

CnCRAZER commented:
For now we only need to remove the animation listing for ra2mode that was attached to the Soviet battle lab, and have the animations after that have their listed index numbers updated. Let's keep the rest of the animations as is

CCCP84 committed:
CRNUKEWH

CCCP84 commented:
<img width="694" height="367" alt="image" src="https://github.com/user-attachments/assets/a3a09143-7e9c-4368-a964-1cd0716ad85d" />

what is indicated in this link, the numbering does not correspond to what is in the files.

There are probably some links to the numbers, although I think it's questionable.
 In any case, I didn't change any numbers. On the contrary, I deleted the added duplicates and left the original numbers.
 I changed the numbering only in the newly added lines

CnCRAZER commented:
> <img alt="image" width="694" height="367" src="https://private-user-images.githubusercontent.com/144725555/515421508-a3a09143-7e9c-4368-a964-1cd0716ad85d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM0MTk3NjAsIm5iZiI6MTc2MzQxOTQ2MCwicGF0aCI6Ii8xNDQ3MjU1NTUvNTE1NDIxNTA4LWEzYTA5MTQzLTdlOWMtNDM2OC1hOTY0LTFjZDA3MTZhZDg1ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTExN1QyMjQ0MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YTI0NGU4MjE2Mjk4MTNiMjhlNzJkOTY1NmFiZmFiMmNhYTk2ZjI5ZmNkMjkzNTk3YTQzMzJiZWEyOTU1YjMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ekix2jWxH2qv4Qqq-D4M4kdizuX6TTnJe96H0nYiou0">
> 
> what is indicated in this link, the numbering does not correspond to what is in the files.
> 
> There are probably some links to the numbers, although I think it's questionable. In any case, I didn't change any numbers. On the contrary, I deleted the added duplicates and left the original numbers. I changed the numbering only in the newly added lines

The games .exe generates an internal array of all the listed animations at load time regardless if they are numbered correctly.

<img width="396" height="328" alt="image" src="https://github.com/user-attachments/assets/87f19329-2fa4-42f0-a4b3-451b462e91f1" />

Deleting the comment for 611 and readding it will add +1 to everything else afterwards and can interfere with campaign missions or custom maps with triggers. 

There is no need at this time to do so

CCCP84 commented:
Well, here's the answer to our question, indicated right in the file.

<img width="528" height="334" alt="image" src="https://github.com/user-attachments/assets/8f81b644-67eb-47cb-8855-eee5b3785e87" />

CCCP84 commented:
but just in case, I can return the commented lines and remove only the duplication of the mininuke. Well, and renumber the newly added lines in a single standard.

Belonit commented:
> In any case, I didn't change any numbers. On the contrary, I deleted the added duplicates and left the original numbers.

The numbers in a ini list don't affect anything. The order of the list is important

CCCP84 commented:
> The numbers in a ini list don't affect anything. The order of the list is important

I see

CnCRAZER commented:
> > The numbers in a ini list don't affect anything. The order of the list is important
> 
> I see

For example, if this is rules file listing:

```
[Animations]
1=TWLT100
3=ELECTRO
;GEF 15=BURN-L
22=H2O_EXP1
23=H2O_EXP2
24=H2O_EXP3
```

When game is launched the .exe file generates a new internal one to this:
```
1=TWLT100
2=ELECTRO
3=H2O_EXP1
4=H2O_EXP2
5=H2O_EXP3
```


But if you delete the comment for `GEF 15=BURN-L` it messes up the list to this:
```
1=TWLT100
2=ELECTRO
3=BURN-L
5=H2O_EXP1
6=H2O_EXP2
7=H2O_EXP3
```


After you would apply this, any and every single map file with any trigger using animation 3 of `H2O_EXP1` (before the change) is now going to use new animation listing of 3 of `BURN-L`. Which would need every map checked and updated accordingly

CCCP84 committed:
Update rulesmd.ini

CCCP84 committed:
Update rulesmd.ini

CCCP84 committed:
Update rulesmd.ini

CCCP84 committed:
Update rulesmd.ini

CCCP84 commented:
I've reworked it so it should be safe now. The duplicate mini nuke has been removed, the value with the forgotten comma has been removed, and the numbers in RA2 and YR now match.

CnCRAZER committed:
Cleanup

CCCP84 commented:
@CnCRAZER Listen, after all this, why did you change the numbering back? I did it so that the numbers would match between RA2 and YR. That was the point. You just went and changed everything back.

CnCRAZER commented:
> @CnCRAZER Listen, after all this, why did you change the numbering back? I did it so that the numbers would match between RA2 and YR. That was the point. You just went and changed everything back.

What's the significance of them matching? This PR was labeled a correction. The incorrect animation being deleted was sufficient

CCCP84 commented:
Because I'm now going to add more animations to the buildings I'm adding, and their numbers are smaller. It's clear that I'll replace this now.  But why should I do this again a second time?
It's just convenient when they match.

CnCRAZER commented:
> Because I'm now going to add more animations to the buildings I'm adding, and their numbers are smaller. It's clear that I'll replace this now. But why should I do this again a second time? It's just convenient when they match.

I am a bit cautious at times regarding any modications to rules.
I knew two things:
1. The only issue was the value that was added by mistake by Belonit
2. Everything else was fine the way it was before any renumbering

For sanity check I wanted it how it was, but with the fix. I at the time did not know of your other PR regarding adding animations

CCCP84 commented:
All ok. It just seems more convenient to when the numbers match, and the numbers really don't matter. So if they match, it will just be convenient.
